src/bun.js/bindings/exports.zig

```diff
@@ -1031,6 +1048,343 @@ pub const ZigConsoleClient = struct {
+                    }
+                }
+            } else if (this.properties.isUndefined()) {
+                // not object -> the value will go to the special "Values" column
+                this.values_col_width = @max(this.values_col_width orelse 1, this.getWidthForValue(row_value));
+            }
+        }
+
+        inline fn writeStringNTimes(writer: anytype, str: []const u8, n: usize) !void {
+            for (0..n) |_| {
+                try writer.writeAll(str);
+            }
+        }
+
+        fn printRow(
+            this: *TablePrinter,
+            writer: anytype,
+            columns: *std.ArrayList(Column),
+            row_key: RowKey,
+            row_value: JSValue,
+        ) !void {
+            try writer.writeAll("│");
+            {
+                const len: u32 = switch (row_key) {
+                    .str => |value| @intCast(value.length()),
+                    .num => |value| @intCast(std.fmt.count("{d}", .{value})),
+                };
+                const pad_l = (columns.items[0].width - len) >> 1;
+                const pad_r = columns.items[0].width - len - pad_l;
+                try writer.writeByteNTimes(' ', pad_l + PADDING);
+                switch (row_key) {
+                    .str => |value| try writer.print("{}", .{value}),
+                    .num => |value| try writer.print("{d}", .{value}),
+                }
+                try writer.writeByteNTimes(' ', pad_r + PADDING);
+            }
+
+            for (1..columns.items.len) |col_idx| {
+                const col = columns.items[col_idx];
+
+                try writer.writeAll("│");
+
+                var value = JSValue.zero;
+                if (col_idx == 1 and this.jstype.isMap()) { // is the "Keys" column, when iterating a Map?
+                    value = row_value.getIndex(this.globalObject, 0);
+                } else if (col_idx == this.values_col_idx) { // is the "Values" column?
+                    if (this.jstype.isMap()) {
+                        value = row_value.getIndex(this.globalObject, 1);
+                    } else if (!row_value.isObject()) {
+                        value = row_value;
+                    }
+                } else if (row_value.isObject()) {
+                    value = row_value.getWithString(this.globalObject, col.name) orelse JSValue.zero;
+                }
+
+                if (value.isEmpty()) {
+                    try writer.writeByteNTimes(' ', col.width + 2 * PADDING);
+                } else {
+                    const len: u32 = this.getWidthForValue(value);
+
+                    const pad_l = (col.width - len) >> 1;
+                    const pad_r = col.width - len - pad_l;
+                    try writer.writeByteNTimes(' ', pad_l + PADDING);
+                    const tag = ZigConsoleClient.Formatter.Tag.get(value, this.globalObject);
+                    switch (Output.enable_ansi_colors) {
+                        inline else => |enable_ansi_colors| {
```

Comment:
IMO, it would be better to do this as a comptime argument to this function and then we only need to check it once instead of each column

but that's a small thing